scripts/lua2dox.lua

```diff
@@ -224,8 +224,8 @@ local function process_magic(line, generics)
   if ty then
     -- fix optional parameters
     if magic_split[2]:find('%?$') then
-      if not ty:find('nil') then
-        ty = ty  .. '|nil'
+      if not ty:find('?') then
+        ty = ty  .. '?'
```

Comment:
This contributed to a lot of the incorrectly generated type. I still have to manually check each type to verify correctness though.

Reply:
This change will affect other generated types in documentation. I only pushed changes for the LSP module though. Should I push changes for _every_ module? I haven't modified non-LSP modules yet. But as I type this, I think it's best to push all docs that are modified by these changes. I'll update the PR.

Reply:
It is a requirement to make sure all docs are updated, CI will fail otherwise.